### PR TITLE
Update tests to use fakes3

### PR DESCRIPTION
**Pull request by tomaszwojcikowski**

### CODE REVIEW

File: test/s3_test.erl [hunk 1]
```diff
@@ -39,10 +38,6 @@ get_put() ->
     ?assertMatch({ok, _}, s3:put(bucket(), <<"foo-copy">>, <<>>, "text/plain",
                                  5000, [{"x-amz-copy-source", bucket() ++ "/foo"}])).
 
-permission_denied() ->
-    ?assertEqual({error, {"AccessDenied", "Access Denied"}},
```
Comment (maxkuzmins) [inline test/s3_test.erl:43]: Has something broke the test?

File: test/s3_test.erl [hunk 2]
```diff
@@ -134,6 +129,10 @@ list_objects() ->
                   s3:list(bucket(), "", 10, "")).
 
 fold() ->
+        {ok, _} = s3:put(bucket(), "1/1", "foo", "text/plain"),
+        {ok, _} = s3:put(bucket(), "1/2", "foo", "text/plain"),
+        {ok, _} = s3:put(bucket(), "1/3", "foo", "text/plain"),
+        {ok, _} = s3:put(bucket(), "2/1", "foo", "text/plain"),
```
Comment (maxkuzmins) [inline test/s3_test.erl:135]: Why these puts became necessary? The tests order doesn't seem to have changed.

Comment (tomaszwojcikowski) [reply test/s3_test.erl:135]: This test depended on the previous test putting objects to s3.

File: test/s3_test.erl (hunk 1 above)
Comment (tomaszwojcikowski) [reply test/s3_test.erl:43]: There is no access checks on fakes3, this test depended on real S3.

**Review by maxkuzmins — APPROVED** [review]

**Pull request merged**